WalletWasabi.Gui/Global.cs

```diff
@@ -156,9 +156,10 @@ public async Task InitializeNoWalletAsync(TerminateService terminateService)
 						await TorManager.StartAsync(ensureRunning: true).ConfigureAwait(false);
 					}
 
-					var fallbackRequestTestUri = new Uri(Config.GetFallbackBackendUri(), "/api/software/versions");
-
-					HostedServices.Register(new TorMonitor(period: TimeSpan.FromSeconds(3), fallbackRequestTestUri, Config.TorSocks5EndPoint, TorManager), nameof(TorMonitor));
+					var httpClient = (WalletWasabi.Tor.Http.TorHttpClient)Synchronizer.HttpClientFactory.NewHttpClient(() => Config.GetFallbackBackendUri(), isolateStream: false);
```

Comment:
There are two things here:

1. `HttpClientFactory` should probably be a property of `Global`.
2. One can avoid the cast by introducing a method in `HttpClientFactory` that returns `TorHttpClient` (or throws an exception if Tor is disabled)

WDYT?

Reply:
I see no reason to move `HttpClientFactory` to `Global`.

Reply:
Well, `HttpClientFactory` should be a property because it is something that is used to initialize other classes. But if you are fine with how it is here now (it's a little bit awkward and I think the property makes better sense but anyway ...) then good.

Reply:
It is as awkward as where it is now - the awkwardity level is equal. I would touch this next time only if DI container arrives and put it into it.

Reply:
@kiminuo now that this was merged can you move the HttpClientFactory to Global, hopefully, the final place of it.

Reply:
Yes.